shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/data/repository/UserRepositoryTest.kt

```diff
@@ -0,0 +1,69 @@
+package co.nimblehq.mark.kmmic.data.repository
+
+import co.nimblehq.mark.kmmic.data.service.user.UserService
+import co.nimblehq.mark.kmmic.data.service.user.model.UserApiModel
+import co.nimblehq.mark.kmmic.domain.model.User
+import co.nimblehq.mark.kmmic.domain.repository.UserRepository
+import kotlin.test.BeforeTest
+import io.kotest.matchers.shouldBe
+import io.mockative.Mock
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import org.koin.core.context.startKoin
+import org.koin.dsl.module
+import kotlin.test.Test
+import kotlinx.coroutines.test.runTest
+import org.koin.core.context.stopKoin
+import kotlin.test.AfterTest
+
+@ExperimentalCoroutinesApi
+class UserRepositoryTest {
+    @Mock
+    private val mockUserService = mock(classOf<UserService>())
+    private lateinit var repository: UserRepository
+    private val mockUser = UserApiModel(
+        "id",
+        "type",
+        "dev@nimblehq.co",
+        "Nimble Team",
+        "https://www.example.com/image.png"
+    )
+
+    @BeforeTest
+    fun setUp() {
+        startKoin{
+            modules(
+                module {
+                    single { mockUserService }
+                }
+            )
+        }
+
+        repository = UserRepositoryImpl()
+    }
+
+    @AfterTest
+    fun tearDown() {
+        stopKoin()
+    }
+
+    @Test
+    fun `when me is called - it returns the correctly user`() = runTest {
+        given(mockUserService)
+            .function(mockUserService::getProfile)
+            .whenInvoked()
+            .thenReturn(
+                flow {
+                    emit(mockUser)
+                }
+            )
+
+        repository.getProfile().collect {
+            it shouldBe User(mockUser)
+        }
```

Comment:
Here's an easy way to assert flow results:
```suggestion
        repository.getProfile().first() shouldBe User(mockUser)
```
We can do the same in other classes too.

Reply:
@AVI5HEK updated at [e796eb2](https://github.com/markgravity/kmm-ic/pull/72/commits/e796eb2998575c9a9fa82a0f9041c6b3db961ce0)